Guides/Transactions.md

```diff
@@ -0,0 +1,111 @@
+# Swift Driver Transactions Guide
+
+`MongoSwift` 1.0.0 added support for [transactions](https://docs.mongodb.com/manual/core/transactions/), which allow applications to use multi-statement transactions that guarantee the atomicity of reads and writes to multiple documents (in a single or multiple collections). Applications can use transactions instead of implementing complex and error-prone ACID-compliant logic themselves, simplifying development and allowing developers to focus on the new features that really matter.
```

Comment:
>which allow applications to use multi-statement transactions

When defining "transactions", we probably shouldn't use the word "transactions" again. This could be rewritten to something like "which allow applications to execute multiple read and write operations atomically across multiple documents and/or collections."

>Applications can use transactions instead of implementing complex and error-prone ACID-compliant logic themselves, simplifying development and allowing developers to focus on the new features that really matter.

I think this is a little too marketing-speaky for our guides. I think it's also important to note, for users who are used to relational databases, that transactions are not always necessary in MongoDB. Maybe we could update it to something milder like: "Transactions reduce the need for complicated application logic when operating on several different documents simultaneously; however, because operations on single documents are always atomic, transactions are often not necessary.", or something like that.

Reply:
Done.